### PR TITLE
fix for helpers serializers tests

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fix for helpers serializers tests

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: aea/helpers/serializers.py [hunk 1]
```diff
@@ -142,13 +144,11 @@ def _patch_dict_restore(cls, dictionary: Dict[str, Any]) -> None:
         # protobuf Struct doesn't recursively convert Struct to dict
         need_patch = dictionary.pop(cls.NEED_PATCH, {})
         for key, value in dictionary.items():
+
             # protobuf struct doesn't recursively convert Struct to dict
             if isinstance(value, Struct):
                 dictionary[key] = cls._restore_value(value)
 
-            if isinstance(value, dict):
```
Comment (5A11) [inline aea/helpers/serializers.py:149]: Why remove restoring nested dicts?

File: tests/test_helpers/test_serializers.py [hunk 2]
```diff
@@ -35,7 +37,13 @@ def test_encode_decode_i():
         "list_of_ints": [1, 2, 3],
         "list_of_str": ["1234", "234234"],
         "list_of_floats": [1.1, 2.2, 3.0],
-        "nested_dict": {"a": b"test", "b": 10, "c": [b"1", b"2"]},
+        "nested_dict": {
+            "a": b"test",
+            "b": 10,
+            "c": [b"1", b"2"],
```
Comment (5A11) [inline tests/test_helpers/test_serializers.py:43]: Please add list of ints, floats, bools, strings as well.

File: tests/test_helpers/test_serializers.py [hunk 3]
```diff
@@ -35,7 +37,13 @@ def test_encode_decode_i():
...
+            "e": {},
+            "d": {"c": 100},
```
Comment (5A11) [inline tests/test_helpers/test_serializers.py:45]: Add `str: None` case.

File: tests/test_helpers/test_serializers.py [hunk 4]
```diff
@@ -35,7 +37,13 @@ def test_encode_decode_i():
...
+        },
         "list_of_dict": {
             "list": [{"a": 1, "b": {"c": 12}}, {"b": b"234", "x": [1, 2, 3]}]
         },
```
Comment (5A11) [inline tests/test_helpers/test_serializers.py:49]: Add a list of a nested dict case as well.

Comment (solarw) [reply tests/test_helpers/test_serializers.py:49]: it's added already

File: tests/test_helpers/test_serializers.py (hunk 2 above)
Comment (solarw) [reply tests/test_helpers/test_serializers.py]: updated

File: tests/test_helpers/test_serializers.py (hunk 4 above)
Comment (solarw) [reply tests/test_helpers/test_serializers.py:57]: updated

File: aea/helpers/serializers.py (hunk 1 above)
Comment (solarw) [reply aea/helpers/serializers.py:149]: cause no dicts provided, only structs

File: tests/test_helpers/test_serializers.py [hunk 5]
```diff
@@ -46,6 +62,15 @@ def test_encode_decode_i():
     assert case == decoded
 
 
+def test_error_type_not_supported():
```
Comment (5A11) [inline tests/test_helpers/test_serializers.py:65]: Is having a list that contains different element types (e.g. bytes, int, float) supported? 
if not, please add it here. If so, add it in the above method.

Comment (solarw) [reply tests/test_helpers/test_serializers.py:65]: good point.
mixed types not supported at the moment. test and explicit check in code are added

**Review by 5A11 — APPROVED** [review]
LGTM

**Pull request merged**